client/src/components/Modal/tests/Modal-test.js

```diff
@@ -0,0 +1,90 @@
+/* global jest, test, describe, it, expect */
+
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import Modal from '../Modal';
+
+/** Component that printouts all of its props */
+const PrintProps = (componentName) => ({ children, ...propsToPrint }) => (
+  <div>
+    <div>{componentName}</div>
+    <ul>
+      {Object
+        .entries(propsToPrint)
+        .map(([key, value]) => (<li key={key}>{key}: {value.toString()}</li>))
+      }
+    </ul>
+    {children}
+  </div>
+);
+
+const makeProps = () => ({
+  isOpen: true,
+  className: 'my-modal',
+  modalClassName: 'my-modal-dialog',
+  size: 'sm',
+  onClosed: jest.fn(),
+  title: 'Hello World!',
+  showCloseButton: true,
+});
```

Comment:
```suggestion
function makeProps(obj = {}) {
  return {
    isOpen: true,
    className: 'my-modal',
    modalClassName: 'my-modal-dialog',
    size: 'sm',
    onClosed: () => {},
    title: 'Hello World!',
    showCloseButton: true,
    ...obj
  };
});
```

I'm not completely sure if its required to prevent bleed over to other tests, though should define each jest.fn() in the test() that is asserting that it was called

Reply:
There's three ways you can avoid bleeding test when using mock functions:
- call `jest.fn()` for each test so you get a different mock
- Call `mockClear()` on the method that got returned from `jest.fn()`
- `jest.clearAllMocks()`

Normally, you would put the `mockClear` or `clearAllMocks` calls in a `beforeEach` or `afterEach` call.